robottelo/entities.py

```diff
@@ -323,19 +334,27 @@ def create_missing(self, auth=None):
         filled in with values too.
 
         """
-        super(ComputeResource, self).create_missing(auth)
         cls = type(self)
+        provider = vars(self)['provider']
```

Comment:
If the user has selected some provider, then these two lines of code are equivalent:

    self.provider
    vars(self)['provider']

The `create_missing` method is responsible for filling in whatever required attributes the user did not provide themselves. If a user does not _not_ select a provider, this will throw an exception due to a dict lookup error:

    vars(self)['provider']

And this will return a `StringField` object:

    self.provider